consul/autopilot.go

```diff
@@ -113,13 +82,18 @@ func (s *Server) pruneDeadServers() error {
 		}
 	}
 
+	// Nothing to remove, return early
+	if len(failed) == 0 {
+		return nil
+	}
+
 	peers, err := s.numPeers()
 	if err != nil {
 		return err
 	}
 
 	// Only do removals if a minority of servers will be affected
-	if len(failed) <= peers/2 {
+	if len(failed) < peers/2 || (len(failed) == 1 && peers >= 3) {
```

Comment:
Are you sure you need this extra case? If we had three servers we don't want to prune until we have 4 (1 dead + 3 alive) - the first case will hit that.